xla/service/gpu/stream_attribute_annotator_test.cc

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2024 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/service/gpu/stream_attribute_annotator.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+#include "tsl/platform/status_matchers.h"
+#include "tsl/platform/statusor.h"
+#include "xla/hlo/ir/hlo_instruction.h"
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/service/gpu/backend_configs.pb.h"
+#include "xla/tests/hlo_test_base.h"
+
+namespace xla {
+namespace gpu {
+namespace {
+
+using StreamAttributeAnnotatorTest = HloTestBase;
+
+TEST_F(StreamAttributeAnnotatorTest, AllUsersAreAnnotated) {
+  constexpr absl::string_view kHloString = R"(
+  HloModule ModuleWithAsync
+
+  ENTRY entry {
+    p1_32 = f32[1] parameter(0)
+    p2_32 = f32[1] parameter(1)
+    add_32 = f32[1] add(p1_32, p2_32), backend_config={"operation_queue_id":"1", "wait_on_operation_queues":[]}
```

Comment:
If we are going to keep this attrs let's make them consistent with names used in runtime:

operation_queue_id -> stream_id (execution_stream_id?)
wait_on_operation_queues -> wait_for_stream_ids

Reply:
Makes sense, this has been used in multiple places now in test files. If we decide to keep it, i will open a NFC PR to change the name first.